### ISSUE TITLE
Toolbar white margin on secondary sites?

### ISSUE THREAD

**Issue by boonebgorges** [cuny-academic-commons/commons-in-a-box#177]
A question I'm passing along from @profstein and the BMCC team. On secondary sites, the toolbar has a white bar underneath:
![screenshot_2019-02-15_13-59-21](https://user-images.githubusercontent.com/246627/52881374-fd9f3380-3129-11e9-9967-0511aa9b01cf.png)

OpenLab team - is this by design?

**Comment by bree-z** [comment]
I was about to put in a ticket for this and found this one! We've discussed this issue recently and would like to port the change we did for this on the OpenLab to CBOX OpenLab.

Here's the OpenLab ticket: http://redmine.citytech.cuny.edu/issues/2645 

When we got rid of this, it revealed another issue, that the OpenLab dropdowns are not aligned with the Nav buttons both horizontally and vertically: http://redmine.citytech.cuny.edu/issues/2667

And in case it's helpful for reference, there was a master ticket for all of the WP toolbar adjustments we made at the same time: http://redmine.citytech.cuny.edu/issues/2545

Thanks!

**Comment by boonebgorges** [comment]
I eliminated the white border.

I also spent some time trying to standardize the CSS treatment of the toolbar. It was littered with weird 1px offsets that made it almost impossible to adjust one thing without breaking everything else. I converted the basic layout to use Flexbox, and I set a standard 50px height for all internal elements on the toolbar. This allowed me to eliminate most of the special cases for styling on toolbar buttons. I'm unsure whether this addresses the "aligned with Nav buttons" issue - I saw some 1px offsets originally (it's what led me on this goose chase) but I'm not sure I fully understand the original report. The changes are available for testing on openlabdev.commonsinabox.org.

**Comment by beckej13820** [comment]
Hi Boone,

I was curious so I checked out a site on the openlabdev.commonsinabox.org and on subsites there might be some unintended consequences. 

https://user-images.githubusercontent.com/29982981/165288182-733a922b-151b-4b2d-aecf-5fad8d67b3dd.mov

You have to scroll down to see it, but instead of a white margin, now there is a black floating line across the site.

**Comment by boonebgorges** [comment]
@beckej13820 The issue in your movie is unrelated to the changes in this ticket. It's an issue specific to the `twentyfifteen` theme, which adds a `fixed` box to the top and bottom of the screen, but offsets it differently based on the presence of admin bars. This doesn't play very nicely with the OpenLab toolbar, and it's something we could potentially address in a theme-specific stylesheet if we had the demand and resources to do so, but in any case it's independent of the current issue. See https://core.trac.wordpress.org/browser/tags/5.9.3/src/wp-content/themes/twentysixteen/style.css#L2782

**Comment by beckej13820** [comment]
Perfect, It's not a problem that I am having, I was just trying to check out the new changes on the dev server.

**Comment by bree-z** [comment]
Thanks, Boone!

I no longer see the white margin with some themes (Twenty Sixteen, Seventeen, Fifteen, Twelve, iKnow), but still see it with others (Jillian, Twenty Thirteen - for example: https://openlabdev.commonsinabox.org/parvi-enim-primo/). Not sure though if these are theme-specific issues.

**Jillian:**
<img width="1440" alt="Screen Shot 2022-05-04 at 4 08 52 AM" src="https://user-images.githubusercontent.com/7189824/166645021-e5bb75f8-9aa3-4fa5-bb12-975ba3602de9.png">

**Twenty Thirteen:**
<img width="1429" alt="Screen Shot 2022-05-04 at 4 06 51 AM" src="https://user-images.githubusercontent.com/7189824/166645076-69bab0a4-277e-4177-8ebd-40f88b1a99a6.png">

**Comment by boonebgorges** [comment]
Thanks for the screenshots. I've pushed another change that should fix this issue.

**Comment by bree-z** [comment]
cc @jennaspevack to follow for any design stuff

**Comment by bree-z** [comment]
This looks good, thanks!

**Comment by profstein** [comment]
Glad to see this moving along. I have some CSS overrides in a mu-plugin on the BMCC OpenLab that addresses some of these issues. It looks like I may be able to remove it once this is in place. We have a handful of sites that use twenty fifteen so I may look into that. Let me know if there is anything else you would like me to look at or share around the admin bar.
